src/mina-signer/src/rosetta.ts

```diff
@@ -49,12 +85,93 @@ function fieldFromHex<T extends Field | Scalar>(
   return [binable.fromBytes(bytes), paddingBit];
 }
 
+function signTransaction(transaction: UnsignedTransaction, privateKey: string, network: NetworkId): SignedRosetta<UnsignedTransaction> {
+  let signature: SignatureJson;
+  if (transaction.payment !== null) {
+    let payment = unsignedTransactionPaymentToPaymentJson(transaction.payment);
+    signature = signPayment(payment, privateKey, network);
+  }
+  else if (transaction.stakeDelegation !== null) {
+    let delegation = unsignedTransactionStakeDelegationToDelegationJson(transaction.stakeDelegation);
+    signature = signStakeDelegation(delegation, privateKey, network);
+  }
+  else {
+    throw Error('signTransaction: Unsupported transaction');
+  }
+  let publicKey = PublicKey.toBase58(PrivateKey.toPublicKey(PrivateKey.fromBase58(privateKey)));
+  return {
+    data: transaction,
+    signature: signatureJsonToHex(signature),
+    publicKey,
+  };
+}
+
+function unsignedTransactionPaymentToPaymentJson(payment: Payment): PaymentJson {
```

Comment:
nit: not super important but could you simplify these long names? e.g. `paymentFromRosetta()`, `delegationFromRosetta()`